src/views/Unsubscribe/index.tsx

```diff
@@ -0,0 +1,177 @@
+                    filterAlertCountryId
+                }
+            }
+        }
+    }
+`;
+
+const UPDATE_SUBSCRIPTION = gql`
+    mutation Unsubscription(
+        $subscriptionId: ID!,
+        $data: UserAlertSubscriptionInput!,
+    ) {
+        private {
+            updateUserAlertSubscription(
+                id: $subscriptionId,
+                data: $data,
+            ) {
+                errors
+                ok
+                result {
+                    id
+                    isActive
+                }
+            }
+        }
+    }
+`;
+
+// eslint-disable-next-line import/prefer-default-export
+export function Component() {
+    const { subscriptionId } = useParams<{ subscriptionId?: string, token?: string }>();
+    const alert = useAlert();
+    const navigate = useNavigate();
+    const strings = useTranslation(i18n);
+    const [isErrored, setIsError] = useState(false);
+
+    const {
+        data: alertSubscriptions,
+    } = useQuery<
+        AlertSubscriptionsUnsubscribeQuery,
+        AlertSubscriptionsUnsubscribeQueryVariables
+    >(
+        ALERT_SUBSCRIPTIONS,
+    );
+
+    const [
+        subscriptionUpdate,
+    ] = useMutation<
+        UnsubscriptionMutation,
+        UnsubscriptionMutationVariables
+    >(
+        UPDATE_SUBSCRIPTION,
+        {
+            onCompleted: (response) => {
+                const subscriptionResponse = response?.private.updateUserAlertSubscription;
+                if (!response) {
+                    return;
+                }
+                if (
+                    subscriptionResponse.ok
+                    && subscriptionResponse.result?.isActive === false
+                ) {
+                    alert.show(
+                        strings.unsubscribeSuccessfully,
+                        { variant: 'success' },
+                    );
+                    navigate('/');
+                } else {
+                    setIsError(true);
+                }
+            },
+            onError: () => {
+                alert.show(
+                    strings.unsubscriptionFailed,
+                    { variant: 'danger' },
+                );
+            },
+        },
+    );
+
+    const data = alertSubscriptions?.private?.userAlertSubscriptions?.items;
+
+    const handleUnsubscribe = useCallback((id: string) => {
+        const selectedSubscriptionDetails = data?.find(
+            (sub) => sub.id === id,
+        );
+
+        if (subscriptionId) {
+            subscriptionUpdate({
+                variables: {
+                    subscriptionId,
+                    data: {
+                        isActive: false,
+                        filterAlertAdmin1s: selectedSubscriptionDetails?.filterAlertAdmin1s ?? [],
+                        filterAlertCountry: selectedSubscriptionDetails?.filterAlertCountryId ?? '',
+                        name: selectedSubscriptionDetails?.name ?? '',
+                    },
+                },
```

Comment:
Instead of performing null checks, we should directly verify if the selected subscription exists and use it. The current logic risks unintentionally modifying the subscription.